utilities/web_ui/base_page.py

```diff
@@ -0,0 +1,55 @@
+from selenium.webdriver.support.wait import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+
+
+class BasePage:
+    def __init__(self, driver):
+        self._driver = driver
+        self.__wait = WebDriverWait(self._driver, 10)
+
+    def __wait_until_element_is_visible(self, locator: tuple):
+        return self.__wait.until(EC.visibility_of_element_located(locator))
+
+    def __wait_until_element_is_clickable(self, locator):
+        return self.__wait.until(EC.element_to_be_clickable(locator))
+
+    def click(self, locator):
+        self.__wait_until_element_is_clickable(locator).click()
+
+    def send_keys(self, locator, value):
+        element = self.__wait_until_element_is_visible(locator)
+        element.send_keys(value)
+
+    def wait_url_to_be(self, url):
+        self.__wait.until(EC.url_to_be(url))
+        return self
```

Comment:
I guess it should return bool

Reply:
@bohdandovbsyh I want to use other methods in the chain after this one. Like this
![Screenshot 2023-09-12 104734](https://github.com/ArtemPurtoian/Python_automation_project/assets/73484419/192d41b7-359d-444a-b951-20fc4a38957c)